config.c

```diff
@@ -2004,7 +2005,7 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                     message(MESS_ERROR, "%s:%d lines must begin with a keyword "
                             "or a filename (possibly in double quotes)\n",
                             configFile, lineNum);
-                    state = STATE_SKIP_LINE;
+                    RAISE_ERROR();
```

Comment:
I think in this case we want to abort reading the whole file and not parse any potential following blocks, so `goto error;` would be more appropriate.

Reply:
> My proposal was actually not to treat unknown configuration directives as errors that would prevent log files from being rotated. logrotate configuration files are often updated independently of logrotate itself and their compatibility with older logrotate releases is important in my opinion.

true, so my comment was void.